### ISSUE TITLE
any options for dev / production mode?

### ISSUE THREAD

**Issue by Kflash** [babel/generator-babel-boilerplate#223]
Hi. Is there any options for dev / production mode inline and under compiling? If not, are you going to add it?

**Comment by megawac** [comment]
What do you want added in production mode?

**Comment by Kflash** [comment]
It all depends on the code you are developing. But I was thinking something like this:

```
if(process.env.NODE_ENV !== 'production') {
    console.info('You\'re running development mode');
}
```
So for dev mode, you can run various stuff depending on your code that shouldn't be available in production mode.

**Comment by megawac** [comment]
Yeah, I was wondering what you want to run includee by default in
development but filtered in prod for this package?

**Comment by Kflash** [comment]
As I said. It all depends on what you develop.  Say you should iterate through an array, and in dev mode you can validate each element in the array and do a console.log etc on it. And in prod mode you only iterate through the array the normal way.

**Comment by jmeas** [comment]
@kflash, this is a useful feature, and one that I often use for my webapps. I think what @megawac is getting at is that there's nothing in the Gulpfile **right now** that could make use of this.

For instance, in my webapps I use `gulp-if` to toggle features like whether or not to uglify the source. In other apps, the determining factor of the environment comes from the server technology, which may or may not be Node, and which may or may not be set up by a JavaScript task runner.

For these reasons, it seems difficult for me to think up a general-purpose way to include this functionality in the boilerplate.

What do you think about adding a Wiki page that describes how to do it, with an example? Do you think that would solve the problem?

**Comment by Kflash** [comment]
@jmeas Sounds like a good idea as long as the wiki will be helpfull enough to include such a feature on my own.

**Comment by Kflash** [comment]
@jmeas Is this going to be added to the readme soon?

**Comment by megawac** [comment]
Feel free to start on a wiki page if you're interested @KFlash

https://github.com/babel/generator-babel-boilerplate/wiki

**Comment by Kflash** [comment]
@megawac I was just wondering about the state on this. It was mentioned a month ago. I abondoned this boilerplate 2 hours ago because it seems stall. Issues reported months ago not fixed etc. And things that are reported deprecated not fixed.

**Comment by jmeas** [comment]
I think I've mentioned this before @kflash, but I work on open source for fun and in my spare time. I'm glad you've found use in this boilerplate, but sometimes way you talk in these issues can come off as sounding entitled.

If you need feature X in open source, then the best way to do that is to contribute to getting it resolved. Otherwise, you're going to have to wait until a maintainer has got time to work on it for you (for free).

I thought that it would make sense for you to start off the Wiki, since this is a feature that you want. Once you get a draft up and running, I can take a look and see if I can make any suggestions to it.

The tl;dr is that I want to help you make the most of this project, but it will work best if we work together on it rather than me doing 100% of the work.

Thanks!

**Comment by jmeas** [comment]
[Done here](https://github.com/babel/generator-babel-boilerplate/wiki/Managing-build-environments)
